### PR TITLE
refactor: using only `update_message_processing_status` to change status

### LINKED ISSUE #189: chore: add Prometheus and Grafana stats for the new queued approach
* Prometheus: is observable for the current number of queued messages to send to relay,
* Grafana: Current backlog time remaining (number / send rate),
* Grafana: How long it took to fully process a /notify call.

This is a follow-up for #141

### PR DESCRIPTION
**Pull request by geekbrother**
# Description

This PR adds a small refactor to use `update_message_processing_status` helper function to change the message status instead of manually running the SQL.
This is useful for tracking metrics inside the `update_message_processing_status` function and following the DRY.

* Making `update_message_processing_status only to change status` to be used with PgPool and Pg transactions,
* Refactor the place of using it.

Resolves #189

## How Has This Been Tested?

Just a build-and-up local smoke test.

## Due Diligence

* [x] 🚧 Stacked on top of #200 
* [ ] Breaking change
* [ ] Requires a documentation update
* [ ] Requires a e2e/integration test update

### CODE REVIEW

File: src/services/publisher_service/helpers.rs [hunk 1]
```diff
@@ -156,12 +153,15 @@ pub async fn pick_subscriber_notification_for_processing(
...
+pub async fn update_message_processing_status<'e, E>(
     notification: Uuid,
     status: SubscriberNotificationStatus,
-    postgres: &PgPool,
+    postgres: E,
     metrics: Option<&Metrics>,
-) -> std::result::Result<(), sqlx::error::Error> {
+) -> std::result::Result<(), sqlx::error::Error>
+where
+    E: sqlx::Executor<'e, Database = sqlx::Postgres>,
```
Comment (chris13524) [inline src/services/publisher_service/helpers.rs:163]: Why this change? Isn't `PgPool` simpler?

File: src/services/publisher_service/helpers.rs [hunk 2]
```diff
@@ -156,12 +153,15 @@ pub async fn pick_subscriber_notification_for_processing(
 }
 
 #[instrument(skip(postgres, metrics))]
-pub async fn update_message_processing_status(
+pub async fn update_message_processing_status<'e, E>(
     notification: Uuid,
     status: SubscriberNotificationStatus,
-    postgres: &PgPool,
+    postgres: E,
```
Comment (chris13524) [inline src/services/publisher_service/helpers.rs:159]: Maybe we could use `PgExecutor` instead to simplify? Looks like fulfills similar purpose as `PgPool`

```suggestion
    postgres: impl PgExecutor<'e>,
```

**Review by chris13524 — APPROVED** [review]
LGTM

File: src/services/publisher_service/helpers.rs (hunk 2 above)
Comment (geekbrother) [reply src/services/publisher_service/helpers.rs]: If it looks more readable, I don't mind to change it ;)
